### PR TITLE
PR to fix ios bgp TC failure

### PR DESCRIPTION
**Pull request by justjais**
Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

##### SUMMARY

PR to fix ios bgp TC failure, as the cleanup was not done properly in case of failure or if the execution was terminated abruptly. 

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

ios_bgp
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by justjais** [comment]
recheck

**Comment by pabelanger** [comment]
The python37 failure is new, looks to be a network failure or appliance problem. But, to be safe we should look to see if our testing didn't cause it. This is the first time I have seen it.

**Comment by pabelanger** [comment]
recheck

**Comment by pabelanger** [comment]
recheck

**Comment by pabelanger** [comment]
@justjais Looking at the code, we are not changing anything in our testing, just making sure we always delete the ios_bgp code we've added. Can you explain more how this will actually fix our flapping tests?

**Comment by justjais** [comment]
recheck

**Comment by justjais** [comment]
recheck

File: test/integration/targets/ios_bgp/tests/cli/basic.yaml [hunk 1]
```diff
@@ -5,390 +5,398 @@
...
+        - 'result.changed == true'
+        - "'router bgp 64496' in result.commands"
+        - "'address-family ipv4 multicast' in result.commands"
+        - "'no redistribute eigrp 233' in result.commands"
+        - "'exit-address-family' in result.commands"
+
+  - name: Configure redistribute information under address family mode with operation replace (idempotent)
+    ios_bgp: *af_rdr_rplc
+    register: result
+    when: ansible_net_version != "15.6(2)T"
```
Comment (pabelanger) [inline test/integration/targets/ios_bgp/tests/cli/basic.yaml:360]: Lets move this 1 task down, as we can still do the idempotent task, just not validate the results.

File: test/integration/targets/ios_bgp/tests/cli/basic.yaml [hunk 2]
```diff
@@ -5,390 +5,398 @@
...
+
+  - assert:
+      that:
+        - 'result.changed == false'
```
Comment (pabelanger) [inline test/integration/targets/ios_bgp/tests/cli/basic.yaml:364]: Or, if we want to keep above, we also need to add a condition check here.

**Review by pabelanger — CHANGES_REQUESTED** [review]

**Review by pabelanger — APPROVED** [review]
Thanks!

**Review by danielmellado — APPROVED** [review]
/lgtm

**Pull request merged**